lib/parse_stringify_toggles.dart

```diff
@@ -5,12 +5,23 @@ import 'package:unleash_proxy_client_flutter/toggle_config.dart';
 /// Parses toggles from the server or local storage so that they can be used
 /// as structured data
 Map<String, ToggleConfig> parseToggles(String body) {
-  final data = jsonDecode(body)['toggles'];
-  // Check if there is anything to map over? Otherwise map might cause an error
-  // Write a test that checks if the
-  return {
-    for (var toggle in data) toggle['name']: ToggleConfig.fromJson(toggle)
-  };
+  var decoded = jsonDecode(body);
+
+  // Return an empty map if 'toggles' isn't a list or doesn't exist
+  if (decoded['toggles'] is! List) {
```

Comment:
I'd really rather avoid checking for a list here. Stuff should fail and the new catch block should catch that, right? I don't really see a reason this function needs to change, it looks like it's correct, it's getting fed garbage data

Reply:
sure makes sense - this was my initial fix, but the try/catch should be enough here agreed

Reply:
Nice! I think this looks good, but maybe we keep this open until we understand more about this problem. I'd need to go root around a little in the code